core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexWaitsIT.java

```diff
@@ -0,0 +1,155 @@
+                                           .get();
+        assertFalse(response.isAcknowledged());
+        deleteIndex(indexName);
+    }
+
+    public void testCreateIndexNotEnoughActiveShardsTimesOut() throws Exception {
+        final String indexName = "test-idx";
+        final int numReplicas = randomIntBetween(2, 4);
+        Settings settings = Settings.builder()
+                                .put(indexSettings())
+                                .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), randomIntBetween(2, 7))
+                                .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), numReplicas)
+                                .build();
+        final int numNodes = randomIntBetween(1, numReplicas - 1);
+        internalCluster().ensureAtMostNumDataNodes(numNodes);
+        CreateIndexResponse response = prepareCreate(indexName)
+                                           .setSettings(settings)
+                                           .setWaitOnActiveShards(numNodes + 1)
+                                           .setTimeout("2s")
+                                           .get();
+        assertFalse(response.isAcknowledged());
+        deleteIndex(indexName);
+    }
+
+    public void testCreateIndexEnoughActiveShards() throws Exception {
+        final String indexName = "test-idx";
+        final int numReplicas = randomIntBetween(2, 4);
+        Settings settings = Settings.builder()
+                                .put(indexSettings())
+                                .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), randomIntBetween(2, 7))
+                                .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), numReplicas)
+                                .build();
+        final int numNodes = randomIntBetween(1, numReplicas + 1);
+        internalCluster().ensureAtMostNumDataNodes(numNodes);
+        CreateIndexResponse response = prepareCreate(indexName)
+                                           .setSettings(settings)
+                                           .setWaitOnActiveShards(randomIntBetween(1, numReplicas + 1))
+                                           .setTimeout("2s")
+                                           .get();
+        assertFalse(response.isAcknowledged());
+        deleteIndex(indexName);
+    }
+
+    public void testCreateIndexWaitsForAllActiveShards() throws Exception {
+        final String indexName = "test-idx";
+        // not enough data nodes, index creation times out
+        final int numReplicas = randomIntBetween(1, 3);
+        Settings settings = Settings.builder()
+                                .put(indexSettings())
+                                .put(INDEX_NUMBER_OF_SHARDS_SETTING.getKey(), randomIntBetween(2, 5))
+                                .put(INDEX_NUMBER_OF_REPLICAS_SETTING.getKey(), numReplicas)
+                                .build();
+        internalCluster().ensureAtMostNumDataNodes(numReplicas);
```

Comment:
will this do anything? The test class specifies that we have exactly 1 data node